Bundler/Viewer/Unity/Assets/Scripts/Persistence/PersistenceSaveLoad.cs

```diff
@@ -112,16 +164,119 @@ public bool SaveLocation(GameObject gameObject, ref WorldAnchorStore store)
             Debug.Log("Attempting save world position\r\n   " + gameObject.transform.localPosition.ToString("0.00"));
 
             // delete any previous WorldAnchors on this object
-            DeleteLocation(gameObject, ref store);
+            DeleteLocation(gameObject, store);
 
-            bool success = false;
+            // add a new anchor to gameObject
+            gameObject.AddComponent<WorldAnchor>();
+
+            // now save the new anchor
+            return SaveExistingLocation(Guid.NewGuid(), gameObject, store);
```

Comment:
If the save fails, the previous location is lost. this should be clearly documented (or the delete should happen after the save)

Reply:
Let me if you're ok with this